addons/sale_timesheet/models/project.py

```diff
@@ -175,6 +176,18 @@ def _compute_sale_line_id(self):
             ], limit=1)
             project.sale_line_id = sol or project.sale_line_employee_ids.sale_line_id[:1]  # get the first SOL containing in the employee mappings if no sol found in the search
 
+    @api.depends('sale_line_id.product_uom_qty', 'sale_line_id.product_uom')
+    def _compute_allocated_hours(self):
+        for project in self:
+            if not project.sale_line_id:
+                project.allocated_hours = 0
+                continue
+            sols = project._get_sale_order_lines()
```

Comment:
Isn't there a way to perform the database accès outside of the loop? I am a bit concerned on a performance prospective.

Reply:
Done